src/auxiliary/p4est.jl

```diff
@@ -13,14 +13,17 @@ This function will check if `p4est` is already initialized
 and if yes, do nothing, thus it is safe to call it multiple times.
 """
 function init_p4est()
-    p4est_package_id = P4est.package_id()
-    if p4est_package_id >= 0
-        return nothing
+    # Only initialize p4est if P4est.jl can be used
+    if P4est.preferences_set_correctly()
```

Comment:
What happens if `p4est` is not initialized here (since their preferences are not set) but a `P4estMesh` is created?

Reply:
First, you see the warning that the preferences are not set correctly. Then it crashes with
```julia
ERROR: LoadError: UndefVarError: `libp4est` not defined
```

Reply:
Should we add a Trixi.jl-specific warning here as well, just to be sure? E.g., something like
```julia
@warn "Preferences for P4est.jl are not set correctly. Until fixed, using `P4estMesh` will result in a crash. See also https://hopefully-not-too-long-link-to-docs"
```
?

Reply:
Sure.

Reply:
Done in [1b02a77](https://github.com/trixi-framework/Trixi.jl/pull/1745/commits/1b02a770727727ec23774acad5c35e6e362c7f67).

Reply:
Great, thanks!